python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -276,10 +277,13 @@ def run(self, splash, version, **kwargs):
         if kwargs.get("server") is None:
             # Initialize all of this after the style-sheet has been applied so any prompt are also
             # styled after the Shotgun Desktop's visual-style.
-            splash.set_message("Initializing browser integration.")
+            if splash:
+                splash.set_message("Initializing browser integration.")
             try:
                 desktop_server_framework = sgtk.platform.get_framework("tk-framework-desktopserver")
-                desktop_server_framework.launch_desktop_server(self._user.host, self._current_login["id"])
+                desktop_server_framework.launch_desktop_server(
+                    self._user.host, self._current_login["id"], parent=splash
```

Comment:
As I mentionned in the other PR, the parenting here is so temporary pop ups while the call is being made are always displayed on top of whatever we are displaying. The framework does not keep a handle to `parent`.